dagger/utilities/dbt_config_parser.py

```diff
@@ -0,0 +1,208 @@
+        prod_dbt_profile = profile_yaml[self._dbt_project_dir.split("/")[-1]]["outputs"]["data"]
+        self._default_data_dir = prod_dbt_profile.get(
+            "s3_data_dir"
+        ) or prod_dbt_profile.get("s3_staging_dir")
+
+    def generate_io(self, model_name: str) -> Tuple[List[Dict], List[Dict]]:
+        """
+        Generates the dagger inputs and outputs for the respective dbt model
+        Args:
+            model_name: name of the dbt model
+
+        Returns:
+            tuple[List[Dict], List[Dict]]: dagger inputs and outputs for the respective dbt model
+
+        """
+        model_parents = self._get_dbt_model_parents(model_name)
+        model_dagger_inputs = self.generate_dagger_inputs(model_parents)
+        model_dagger_outputs = self.generate_dagger_outputs(
+            model_parents["model_name"],
+            model_parents["schema"],
+            model_parents["relative_s3_path"],
+        )
+
+        return model_dagger_inputs, model_dagger_outputs
+
+    def parse_dbt_staging_model(self, dbt_staging_model: str) -> Tuple[str, str]:
+        """
+        Parses the dbt staging model to get the core schema and table name
+        Args:
+            dbt_staging_model: name of the DBT staging model
+
+        Returns:
+            Tuple[str, str]: core schema and table name
+
+        >>> parse_dbt_staging_model("schema_name__table")
+        ('schema_name', 'table')
+        >>> parse_dbt_staging_model("another_schema__another_table")
+        ('another_schema', 'another_table')
+        """
+        _model_split, core_table = dbt_staging_model.split("__")
+        core_schema = _model_split.split("_")[-1]
+
+        return core_schema, core_table
+
+    def generate_dagger_inputs(
+        self, dbt_model_parents: dict
+    ) -> Union[List[Dict], None]:
+        """
+        Generates the dagger inputs for the respective dbt model. This means that all parents of the dbt model are added as dagger inputs.
+        Staging models are added as Athena inputs and core models are added as Athena and S3 inputs.
+        Intermediate models are not added as an input.
+        Args:
+            dbt_model_parents: All parents of the dbt model
+
+        Returns:
+            Union[List[Dict], None]: dagger inputs for the respective dbt model. If there are no parents, returns None
+
+        """
+        dagger_inputs = []
+        for parent in dbt_model_parents["inputs"]:
+            model_name = parent["model_name"]
+            athena_input = ATHENA_IO_BASE.copy()
+            s3_input = S3_IO_BASE.copy()
+
+            if model_name.startswith("stg_"):
+                athena_input["name"] = model_name
+                (
+                    athena_input["schema"],
+                    athena_input["table"],
+                ) = self.parse_dbt_staging_model(model_name)
+
+                dagger_inputs.append(athena_input)
+            else:
+                athena_input["name"] = athena_input["table"] = model_name
```

Comment:
In DBT, we cannot have two models that have the same filename(table/model name) that belong to a different schema. 

But, I get your point. If this could ever happen, it will definitely be a collision. Will make it a combination of schema+table_name to be safe 👍🏽